src/stream.rs

```diff
@@ -196,8 +204,22 @@ impl Compress {
     /// Creates a new compressor ready to encode data into brotli
     pub fn new(params: &CompressParams) -> Compress {
         unsafe {
-            let state = brotli_sys::RustBrotliCompressorCreate(params.params);
+            let state = brotli_sys::BrotliEncoderCreateInstance(None, None, 0 as *mut _);
             assert!(!state.is_null());
+
+            brotli_sys::BrotliEncoderSetParameter(state,
+                                                  brotli_sys::RUST_PARAM_MODE,
+                                                  params.mode_as_native());
+            brotli_sys::BrotliEncoderSetParameter(state,
+                                                  brotli_sys::RUST_PARAM_QUALITY,
+                                                  params.quality);
+            brotli_sys::BrotliEncoderSetParameter(state,
+                                                  brotli_sys::RUST_PARAM_LGWIN,
+                                                  params.lgwin);
+            brotli_sys::BrotliEncoderSetParameter(state,
+                                                  brotli_sys::RUST_PARAM_LGBLOCK,
+                                                  params.lgblock);
```

Comment:
I could do that if you prefer, but I can't see the win to be honest.  Won't most users avoid the `Compress` struct altogether and go straight for the `compress_buf` or `compress_vec` functions?